Objects/listobject.c

```diff
@@ -594,6 +595,13 @@ list_repr_impl(PyListObject *v)
     /* Do repr() on each element.  Note that this may mutate the list,
        so must refetch the list size on each iteration. */
     for (Py_ssize_t i = 0; i < Py_SIZE(v); ++i) {
+        item = list_get_item_ref(v, i);
+        if (item == NULL) {
```

Comment:
Is it possible that list can be truncated here? this function is called with the critical section being held so no  other thread can modify the list, I think a simple incref here would have been sufficient here.

Reply:
I wasn't sure so I picked the safe list_get_item_ref(), but it seems like you're right. I  wrote https://github.com/python/cpython/pull/132811 to just use INCREF.